src/core/server/integration_tests/saved_objects/migrations/group2/multi_node_split.test.ts

```diff
@@ -0,0 +1,292 @@
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { join } from 'path';
+import { omit, sortBy } from 'lodash';
+import type { TestElasticsearchUtils } from '@kbn/core-test-helpers-kbn-server';
+import type { MigrationResult } from '@kbn/core-saved-objects-base-server-internal';
+import type { Client } from '@elastic/elasticsearch';
+import {
+  clearLog,
+  defaultKibanaIndex,
+  defaultKibanaTaskIndex,
+  getAggregatedTypesCount,
+  getEsClient,
+  nextMinor,
+  startElasticsearch,
+} from '../kibana_migrator_test_kit';
+import {
+  BASELINE_COMPLEX_DOCUMENTS_500K_AFTER,
+  BASELINE_DOCUMENTS_PER_TYPE_500K,
+  BASELINE_TEST_ARCHIVE_500K,
+} from '../kibana_migrator_archive_utils';
+import {
+  getRelocatingMigratorTestKit,
+  kibanaSplitIndex,
+} from '../kibana_migrator_test_kit.fixtures';
+import { delay, parseLogFile } from '../test_utils';
+import '../jest_matchers';
+import { expectDocumentsMigratedToHighestVersion } from '../kibana_migrator_test_kit.expect';
+
+const PARALLEL_MIGRATORS = 4;
+type Job<T> = () => Promise<T>;
+
+const getLogFile = (node: number) => join(__dirname, `multi_node_split_${node}.log`);
+const logFileSecondRun = join(__dirname, `multi_node_split_second_run.log`);
+
+describe('multiple Kibana nodes performing a reindexing migration', () => {
+  let esServer: TestElasticsearchUtils['es'];
+  let jobs: Array<Job<MigrationResult[]>>;
+  let client: Client;
+  let results: MigrationResult[][];
+
+  beforeEach(async () => {
+    for (let i = 0; i < PARALLEL_MIGRATORS; ++i) {
+      await clearLog(getLogFile(i));
+    }
+    await clearLog(logFileSecondRun);
+
+    esServer = await startElasticsearch({ dataArchive: BASELINE_TEST_ARCHIVE_500K });
+    jobs = await createMigratorJobs(PARALLEL_MIGRATORS);
+    client = await getEsClient();
+    await checkBeforeState();
+  });
+
+  it('migrate saved objects normally when started at the same time', async () => {
+    results = await startWithDelay(jobs, 0);
+  });
+
+  it('migrate saved objects normally when started with a small interval', async () => {
+    results = await startWithDelay(jobs, 1);
+  });
+
+  it('migrate saved objects normally when started with an average interval', async () => {
+    results = await startWithDelay(jobs, 5);
+  });
+
+  it('migrate saved objects normally when started with a bigger interval', async () => {
+    results = await startWithDelay(jobs, 20);
+  });
+
+  afterEach(async () => {
+    checkMigratorsResults();
+    await checkIndicesInfo();
+    await checkSavedObjectDocuments();
+    await checkMigratorsSteps();
+    await checkUpToDateOnRestart();
+  });
+
+  afterEach(async () => {
+    await esServer?.stop();
+    await delay(5); // give it a few seconds... cause we always do ¯\_(ツ)_/¯
+  });
```

Comment:
2 `afterEach` where `esServer` can be stopped before completing the checks in the other `afterEach`.

Reply:
Oh, that's a good catch, will put them together + `finally` block.